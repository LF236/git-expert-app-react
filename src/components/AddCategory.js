import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [ inputValue, setInputValue ] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.length > 2) {
            setCategories((cat => [  inputValue, ...cat ]));
            setInputValue( '' );
        }       
        // console.log( 'Disparando handleSubmit()' );
    }

    const handleChangeInput = (e) => {        
        setInputValue( e.target.value );
        // console.log( 'HandleInputChange llamdo' );
    }


    return(
        <form onSubmit={ handleSubmit }>
            <span>{ inputValue }</span>
            <br/>
            <input            
                value={ inputValue }
                onChange={ handleChangeInput }                
            />
        </form>

    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;