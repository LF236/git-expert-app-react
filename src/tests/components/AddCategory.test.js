import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe( 'Pruebas sobre el componente <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    } );

    test( 'El componente renderizado, debe ser igual a su Snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'Debe cambiar la caja de texto', () => {
        let value = 'Hello World';
        wrapper.find( 'input' ).simulate( 'change', { target: { value } } );
        // Buscamos el span para verificar que sea igual al inputValue // Lo que cambia en el evento
        expect( wrapper.find( 'span' ).text().trim() ).toBe( value );
    } );

    test( 'Al hacer el submit, no se debe disparar la funciÃ³n setCategories()', () => {
        wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );
        expect( setCategories ).not.toHaveBeenCalled();
    } );

    test( 'Se debe llamar el setCategories y limpiar la caja de texto', () => {
        wrapper.find( 'input' ).simulate( 'change', { target: { value: 'lf236' } } );
        wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes( 1 );
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );
        expect( wrapper.find( 'input' ).prop( 'value' ) ).toBe( '' );
    } );
} );