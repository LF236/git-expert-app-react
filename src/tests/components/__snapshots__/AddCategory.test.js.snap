// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <AddCategory /> El componente renderizado, debe ser igual a su Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <span />
  <br />
  <input
    onChange={[Function]}
    value=""
  />
</form>
`;
